### ISSUE TITLE
Custom stream protocols will sometimes give empty or partial responses (Electron 4)

### ISSUE THREAD

**Issue by pfrazee** [electron/electron#16213]
* Output of `node_modules/.bin/electron --version`: 4.0.0
* Operating System (Platform and Version): MacOS 10.12
* Output of `node_modules/.bin/electron --version` on last known working Electron version (if applicable): 3.0.9

**Expected Behavior**

Custom stream protocol responses should consistently provide the full response body of the request.

**Actual behavior**

Custom stream protocol responses will occasionally give an empty or partial response.

In both cases, the headers are present in devtools. When an empty response is given, the response body is missing.

When a partial response is given, the total size is consistently 4 bytes less than the sum of some of the chunks. For instance, if there are two chunks of 100b then 50b, the received size might be 96b (4b less than the first chunk's size). If there are three chunks of 100b, then 50b, then 25b, the received size might be 96b (4b less than the first chunk) or it might be 146b (4b less than the first two chunks combined). I mention this because the 4b pattern is consistent, and might give a clue about the cause.

I have been unable to reliably reproduce the issue, making me think this is simply a race within the stream-protocol code.

Things which did not seem to affect the behavior:

 - Sandboxing
 - Caching
 - Webview vs running within the browser window
 - Any of the security settings or optional behavior flags on BrowserWindow/WebView
 - The presence of preload scripts
 - The size of the response
 - The timing if the response
 - Which resource was requested (it varies)

**To Reproduce**

```bash
# Clone this repository
git clone https://github.com/pfrazee/electron-bug-test-truncated-protocol-responses
# Go into the repository
cd electron-bug-test-truncated-protocol-responses
# Install dependencies
npm install
# Run the app
npm start
```

Then:

- The window will open with README.md rendered and devtools open. Switch to the network tab so that you can log requests. 
- Typically the request will succeed. Refresh repeatedly (`cmd/ctrl + r`) until the window becomes blank.
- Notice in the network tab, the request response is empty, though the headers are set.

**Comment by vladimiry** [comment]
Related unresolved issues:
- https://github.com/electron/electron/issues/13519
- https://github.com/electron/electron/issues/16192

**Comment by burtonator** [comment]
This is a pretty severe regression on 4.0... a main API should not simply stop working and it shouldn't take weeks to resolve. 

Is this API not tested?  How did the 4.0 release even happen?

It makes it hard to commit to Electron when major platform APIs simply don't work.

**Comment by KoenLav** [comment]
By the way; we are running on Windows and the issue seems the same, so I don't think this is platform specific.

**Comment by jkleinsc** [comment]
Fixed via #16532.

**Comment by pfrazee** [comment]
Awesome, thanks!!
